src/main/java/org/broadinstitute/hellbender/utils/samples/MendelianViolation.java

```diff
@@ -49,14 +57,24 @@ public int getParentsRefRefChildHet() {
         return inheritance.get(GenotypeType.HOM_REF).get(GenotypeType.HOM_REF).get(GenotypeType.HET);
     }
 
-    public boolean isViolation(final Sample mother, final Sample father, final Sample child, final VariantContext vc){
-        violations_total=0;
+    private void resetCounts() {
+        nocall = 0;
+        lowQual = 0;
+        familyCalled = 0;
+        varFamilyCalled = 0;
+        violations_total = 0;
+        violationFamilies.clear();
+
         clearInheritanceMap();
+    }
+
+    public boolean isViolation(final Sample mother, final Sample father, final Sample child, final VariantContext vc){
+        resetCounts();
         updateViolations(mother.getFamilyID(), mother.getID(), father.getID(), child.getID(),vc);
-        return violations_total>0;
+        return violations_total > 0;
     }
 
-    private void updateViolations(final String familyId, final String motherId, final String fatherId, final String childId, final VariantContext vc){
+    protected void updateViolations(final String familyId, final String motherId, final String fatherId, final String childId, final VariantContext vc){
```

Comment:
As I said in the earlier comment, rather than re-implement all of the VariantEval-specific code in MendelianViolation (it tracked various inheritance-related values which as not really MV-related), I am planning to subclass MendelianViolation in MendelianViolationEvaluator (a VariantEval class).  Once VariantEval is merged this will be important.